lib/node_modules/@stdlib/repl/lib/syntax_highlighter.js

```diff
@@ -0,0 +1,154 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable no-underscore-dangle, no-restricted-syntax, no-invalid-this */
+
+'use strict';
+
+// MODULES //
+
+var readline = require( 'readline' );
+var logger = require( 'debug' );
+var setNonEnumerableReadOnly = require( '@stdlib/utils/define-nonenumerable-read-only-property' );
+var tokenizer = require( './tokenizer.js' );
+var THEMES = require( './themes.js' );
+var ANSI = require( './ansi_colors.js' );
+
+
+// VARIABLES //
+
+var debug = logger( 'repl:syntax-highlighter' );
+
+
+// MAIN //
+
+/**
+* Constructor for creating a syntax-highlighter.
+*
+* @private
+* @constructor
+* @param {repl} repl - REPL instance
+* @param {WritableStream} ostream - writable stream
+* @returns {SyntaxHighlighter} syntax-highlighter instance
+*/
+function SyntaxHighlighter( repl, ostream ) {
+	if ( !( this instanceof SyntaxHighlighter ) ) {
+		return new SyntaxHighlighter( repl, ostream );
+	}
+	debug( 'Creating a new syntax-highlighter' );
+
+	// Cache a reference to the provided REPL instance:
+	this._repl = repl;
+
+	// Cache a reference to readline interface:
+	this._rli = repl._rli;
+
+	// Cache a reference to the output writable stream:
+	this._ostream = ostream;
+
+	// Cache a reference to the available themes:
+	this._themes = THEMES;
```

Comment:
Not clear why you need to cache a reference to available themes. The variable is available in the parent scope.

Reply:
We will also have user defined themes that they will giving us through `opts`, and we would have to cache that during initialization of the syntax highlighter. But yes for now, caching themes doesn't make sense, will revert..

Reply:
Similar to the default theme, may make sense to query the set of themes prior to each keypress, as the set of themes may change.

Reply:
As far as I know, the user will only be able to add themes through `opts` during REPL instantiation right? Adding a setting to add new themes is a bit overkill in my opinion (if that's what you are reaching at?)

Reply:
I would think that themes could be programmatically set similar to how settings can be programmatically updated on the REPL instance. In which case, the set of themes may not be fixed after instantiation. Similarly, from within the REPL, I may want to dynamically load a new theme from the local file system, etc. In which case, I don't think that `opts` would be the only entry point.

Reply:
E.g.,

```
In [1]: loadTheme( '/path/to/theme/file.json' )
```